Makefile

```diff
@@ -124,21 +135,25 @@ $(C_DIR_LIB):
 
 $(C_DIR_INCLUDE):
 	mkdir -p $(C_DIR_INCLUDE)
-	mkdir $(C_DIR_INCLUDE)/qiskit
+$(C_DIR_INCLUDE_NAMESPACE):
+	mkdir -p $(C_DIR_INCLUDE_NAMESPACE)
 
 $(C_LIBQISKIT): $(C_DIR_LIB)  $(C_LIB_CARGO_PATH)
 	cp $(C_LIB_CARGO_PATH) $(C_DIR_LIB)/$(subst _cext,,$(C_LIB_CARGO_FILENAME))
 
-$(C_QISKIT_H): $(C_DIR_INCLUDE) $(C_LIB_CARGO_PATH)
-	cp target/qiskit.h $(C_DIR_INCLUDE)/qiskit.h
-	cp crates/cext/include/complex.h $(C_DIR_INCLUDE)/qiskit/complex.h
+target/qiskit.h: $(C_LIBQISKIT)
+$(C_QISKIT_H): $(C_DIR_INCLUDE) $(C_LIB_CARGO_PATH) target/qiskit.h
+	cp target/qiskit.h $(C_QISKIT_H)
+
+$(C_QISKIT_NAMESPACED_H): $(C_DIR_INCLUDE_NAMESPACE) $(C_DIR_SRC_INCLUDE)/$(notdir $@)
+	cp $(C_DIR_SRC_INCLUDE)/$(notdir $@) $(C_DIR_INCLUDE_NAMESPACE)/$(notdir $@)
 
 .PHONY: c cheader
-cheader: $(C_QISKIT_H)
-c: $(C_LIBQISKIT) $(C_QISKIT_H)
+cheader: $(C_QISKIT_H) $(C_QISKIT_NAMESPACED_H)
+c: $(C_LIBQISKIT) cheader
 
 # Use ctest to run C API tests
-ctest: $(C_DIR_INCLUDE)
+ctest: $(C_DIR_INCLUDE)/qiskit
 	cargo rustc --crate-type cdylib -p qiskit-cext
 	cp target/qiskit.h $(C_DIR_INCLUDE)/qiskit.h
 	cp crates/cext/include/complex.h $(C_DIR_INCLUDE)/qiskit/complex.h
```

Comment:
This isn't correct (I'm not sure why I made the one change I did), but right now the `Makefile` is not organised well enough to conditionally choose whether to include the `--release` flag or not, so I've left sorting that out for a follow-up.